nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -0,0 +1,718 @@
+		}
+
+		if preUpgradeGroupID != "" {
+			err = upgradeClientSet.GroupClient.Reorder(groupID, getReorderAfterReq(preUpgradeGroupID))
+			if err != nil {
+				return err
+			}
+		}
+		preUpgradeGroupID = groupID
+	}
+	return nil
+}
+
+func updateComponentUpgradePlanSetting(settingClient plan.SettingsClient, d *schema.ResourceData, component string) error {
+	settingI := d.Get(componentToSettingKey[component]).([]interface{})
+	if len(settingI) == 0 {
+		return nil
+	}
+
+	settingGet, err := settingClient.Get(component)
+	if err != nil {
+		return err
+	}
+
+	setting := settingI[0].(map[string]interface{})
+	parallel := setting["parallel"].(bool)
+	settingGet.Parallel = &parallel
+
+	// PauseOnError can't be modified for EDGE upgrade setting
+	if component != edgeUpgradeGroup {
+		stopOnErr := setting["stop_on_error"].(bool)
+		settingGet.PauseOnError = &stopOnErr
+	}
+
+	_, err = settingClient.Update(component, settingGet)
+	return err
+}
+
+func runUpgrade(upgradeClientSet *upgradeClientSet, partialUpgradeMap map[string]bool) error {
+	partialUpgradeExist := false
+	for i := range upgradeComponentList {
+		// After one component upgrade is completed, although the status of our next component is NOT_STARTED,
+		// there is a period that overall status is still IN_PROGRESS, which will prevent us to start the upgrade of next component.
+		// Wait here for the overall status become stable. Because there is potential upgrade triggered before, we wait here also
+		// for the first component for safety.
+		err := waitUpgradeForStatus(upgradeClientSet, nil, inFlightComponentUpgradeStatus, staticComponentUpgradeStatus)
+		if err != nil {
+			return err
+		}
+
+		component := upgradeComponentList[i]
+
+		if component == mpUpgradeGroup && partialUpgradeExist {
+			log.Printf("[INFO] Some UpgradeUnitGroups haven't been upgraded. MP upgrade is skipped")
+			continue
+		}
+		pendingStatus := []string{model.ComponentUpgradeStatus_STATUS_IN_PROGRESS}
+		targetStatus := []string{model.ComponentUpgradeStatus_STATUS_SUCCESS}
+		completeLog := fmt.Sprintf("[INFO] %s upgrade is completed.", component)
+		if partialUpgradeMap[component] {
+			// For partial upgrade, some groups are disabled, component upgrade status will be paused after all enabled groups upgraded.
+			pendingStatus = append(pendingStatus, model.ComponentUpgradeStatus_STATUS_PAUSING)
+			targetStatus = append(targetStatus, model.ComponentUpgradeStatus_STATUS_PAUSED)
+			partialUpgradeExist = true
+			completeLog = fmt.Sprintf("[INFO] %s upgrade is partially completed.", component)
+		}
+		upgradeClientSet.PlanClient.Upgrade(&component)
+		err = waitUpgradeForStatus(upgradeClientSet, &component, pendingStatus, targetStatus)
+		if err != nil {
+			return err
+		}
+		log.Printf(completeLog)
+	}
+	return nil
+}
+
+func runPostcheck(upgradeClient nsx.UpgradeClient, d *schema.ResourceData) {
+	for i := range upgradeComponentList {
+		component := upgradeComponentList[i]
+		if component != mpUpgradeGroup {
```

Comment:
there are no post upgrade checks for MP?

Reply:
Yes, as you can see there is no post-check btn on UI.